cs/test/core/TypeAliasTests.cs

```diff
@@ -107,6 +108,130 @@ public static ArraySegment<byte> Convert(byte[] value, ArraySegment<byte> unused
         {
             return new ArraySegment<byte>(value);
         }
+
+        #region Box<T> for basic types
```

Comment:
We'd probably want to constrain T anyway but I don't think C#'s current generic constraints are expressive enough for what we need at the moment.